server/routers/api-router.js

```diff
@@ -214,6 +215,67 @@ router.get("/api/status-page/heartbeat", cache("5 minutes"), async (_request, re
     }
 });
 
+router.get("/api/badge/:id/:type", cache("5 minutes"), async (request, response) => {
+    allowDevAllOrigin(response);
+
+    const {
+        label,
+        labelPrefix,
+        labelSuffix = "h",
+        prefix,
+        suffix,
+    } = request.query;
+
+    try {
+        await checkPublished();
+
+        const requestedMonitorId = parseInt(request.params.id, 10);
+        const requestedType = parseInt(request.params.type, 10) ?? 24;
+
+        let publicMonitor = await R.getRow(`
+                SELECT monitor_group.monitor_id FROM monitor_group, \`group\`
+                WHERE monitor_group.group_id = \`group\`.id
+                AND monitor_group.monitor_id = ?
+                AND public = 1
+            `,
+        [requestedMonitorId]
+        );
+
+        const badgeValues = {};
+
+        if (!publicMonitor) {
+            // return a "n/a" badge in grey, if monitor is not public / not available / non exsitant
+
+            badgeValues.message = "N/A";
+            badgeValues.color = "#999";
```

Comment:
Can we use a constant for this and for `percentageToColor()` from `util-server.js`?

Reply:
Good idea! I will update the code